test/src/unit-capi-smoke-test.cc

```diff
@@ -215,19 +216,8 @@ class SmokeTestFx {
   /** The C-API VFS object. */
   tiledb_vfs_t* vfs_;
 
-  /**
-   * Creates a directory using `vfs_`.
-   *
-   * @param path The directory path.
-   */
-  void create_dir(const string& path);
-
-  /**
-   * Removes a directory using `vfs_`.
-   *
-   * @param path The directory path.
-   */
-  void remove_dir(const string& path);
+  /** The unique directory object. */
+  std::optional<UniqueDirectory> temp_dir_{std::nullopt};
```

Comment:
See below for why this should not be `optional` and what the problems are with fixtures.